### DISCUSSION TITLE
Stats Widget Filter

### DISCUSSION THREAD

**Discussion by KarimFekry8** [filamentphp/filament#8246] [Help]
**Is there any way I can have a filter for the Stats widget**?

use Filament\Widgets\StatsOverviewWidget as BaseWidget;
use Filament\Widgets\StatsOverviewWidget\Card;
use Illuminate\Database\Eloquent\Model;

class ExpenseOverview extends BaseWidget
{
    public ?Model $record = null;

    protected function getCards(): array
    {
        return [
            Card::make('Total Expenses', function () {
                $car= $this->record;
                return $car->expenses->sum('amount');
            }),
            Card::make('Month Expenses', function () {
                $car= $this->record;

                $expensesForCurrentMonth = $car->expenses()
                ->whereYear('created_at', now()->year)
                ->whereMonth('created_at', now()->month)
                ->get();

                // Calculate the sum of 'amount' for the filtered expenses
                $totalExpensesForCurrentMonth = $expensesForCurrentMonth->sum('amount');

                return $totalExpensesForCurrentMonth;
            })->description('Expenses for the current month'),
        ];
    }
}

I want to filter by month instead of making the second card.

Thanks

**Comment by danharrin** [comment]
https://laraveldaily.com/post/filament-one-filter-for-multiple-widgets-livewire-events
